manual/api/vector_search.md

```diff
@@ -0,0 +1,221 @@
+## Vecoter Search
+We have developed vector search functionality as an extension for SeaSearch. The following introduces the related APIs.
+
+### Creating a Vector Index
+To use the vector search feature, you must create a vector index. This can be done via mapping.
+
+For example, we create an index and set the vector field in the document data to be written as "vec". The index type is set to `flat` and the vector dimension is `768`.
+
+```
+[PUT] /es/${indexName}/_mapping
+
+body:
+{
+  "properties": {
+    "vec": {
+      "type": "vector",
+      "dims": 768,
+      "vec_index_type": "flat"
+    }
+  }
+}
+```
+
+#### Parameter Description:
+- ${indexName}: The name of the index.
+- type: Fixed as `vector`, indicating that this is a vector index.
+- dims: The vector dimension.
+- m: A parameter required for the `ivf_pq` index, it must be divisible by dims. For example, the `dims` is `768`, the `m` could be `192`.
+- nbits: A parameter required for the `ivf_pq` index, must grater than 0, we recommend `4` or `8`.
+- vec_index_type: The index type. Supported types are `flat` and `ivf_pq`.
+
+### Indexing Documents Containing Vectors
+
+Indexing a document that includes vectors is identical at the API level to indexing a regular document. 
+You may choose the method that best suits your needs. The following example uses the bulk API.
+
+```
+[POST] /es/_bulk
+
+{ "index" : { "_index" : "index1" } }
+{"name": "jack1", "vec": [10.2, 10.41, 9.5, 22.2]}
+{ "index" : { "_index" : "index1" } }
+{"name": "jack2", "vec": [10.2, 11.41, 9.5, 22.2]}
+{ "index" : { "_index" : "index1" } }
+{"name": "jack3", "vec": [10.2, 12.41, 9.5, 22.2]}
+```
+
+### Searching Vectors
+
+You can search the system for the top N similar vectors and return the corresponding document information.
+
+```
+[POST] /api/${indexName}/_search/vector
+
+{
+  "query_field": "vec",
+  "k": 7,
+  "return_fields": ["name"],
+  "vector": [10.2, 10.40, 9.5, 22.2, ...],
+  "_source": false
+}
+```
+The API response format is the same as that for full-text search.
+
+#### Parameter Description
+- ${indexName}: The name of the index.
+- query_field: The field in the index to search. This field must be of type vector.
+- k: The number of most similar vectors to return.
+- return_fields: The names of the fields to return separately.
+- vector: The vector used for querying.
+- nprobe: Applicable only to the `ivf_pq` index type; it specifies the number of clusters to search. The higher the number, the more accurate the results.
+- _source: Controls whether to return the `_source` field. It supports a boolean value or an array specifying which fields to return.
+
+### Recall Query
+For vectors of the `ivf_pq` type, you can perform a recall check on the data. You can optimize and adjust your `ivf_pq` index parameters based on the results of this API.
+It returns a floating point number between 0 and 1, representing the recall rate. The closer it is to 1, the more accurate the search results are.
```

Comment:
For vector indexes of the `ivf_pq` type, you can evaluate the recall for search on the data.